charts/consul/templates/tls-init-job.yaml

```
@@ -1,5 +1,6 @@
 {{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
 {{- if (and .Values.global.tls.enabled (not .Values.server.serverCert.secretName)) }}
+{{- if not (and .Values.global.secretsBackend.vault.enabled .Values.global.tls.enabled) }}
```

```suggestion
{{- if not .Values.global.secretsBackend.vault.enabled }}
```

nit: the `{{- if (and .Values.global.tls.enabled (not .Values.server.serverCert.secretName)) }}` check above covers whether `global.tls.enabled` is set.